src/RuleSet/DeclarationBlock.php

```diff
@@ -13,22 +15,48 @@
 use Sabberworm\CSS\Parsing\UnexpectedTokenException;
 use Sabberworm\CSS\Property\KeyframeSelector;
 use Sabberworm\CSS\Property\Selector;
+use Sabberworm\CSS\Renderable;
+use Sabberworm\CSS\Rule\Rule;
 
 /**
- * This class represents a `RuleSet` constrained by a `Selector`.
+ * This class includes a `RuleSet` constrained by a `Selector`.
  *
  * It contains an array of selector objects (comma-separated in the CSS) as well as the rules to be applied to the
  * matching elements.
  *
  * Declaration blocks usually appear directly inside a `Document` or another `CSSList` (mostly a `MediaQuery`).
  */
-class DeclarationBlock extends RuleSet
+class DeclarationBlock implements Commentable, Renderable
 {
     /**
      * @var array<Selector|string>
      */
     private $selectors = [];
 
+    /**
+     * @var RuleSet
+     */
+    private $ruleSet;
+
+    /**
+     * @var int<0, max>
+     */
+    private $lineNumber;
```

Comment:
… and maybe we can introduce the traits for `Commentable` and for things with a line number first.